spring-cloud-gcp-examples/spring-cloud-gcp-pubsub-example/src/main/java/com/example/WebController.java

```diff
@@ -0,0 +1,95 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.example;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.gcp.pubsub.PubSubAdmin;
+import org.springframework.cloud.gcp.pubsub.core.PubSubTemplate;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.view.RedirectView;
+
+@RestController
+public class WebController {
+
+	@Autowired
+	private PubSubTemplate pubSubTemplate;
+
+	@Autowired
+	private PubSubAdmin pubSubAdmin;
+
+	private static Log LOGGER = LogFactory.getLog(PubSubApplication.class);
+
+	@PostMapping("/createTopic")
+	public RedirectView createTopic(@RequestParam("topicName") String topicName) {
+		this.pubSubAdmin.createTopic(topicName);
+
+		return new RedirectView("/");
+	}
+
+	@PostMapping("/createSubscription")
+	public RedirectView createSubscription(@RequestParam("topicName") String topicName,
+			@RequestParam("subscriptionName") String subscriptionName) {
+		this.pubSubAdmin.createSubscription(subscriptionName, topicName);
+		this.pubSubTemplate.subscribe(subscriptionName,
```

Comment:
You say `createSubscription`, but why you `subscribe` here as well. Unconditionally.
Thanks